src/aind_metadata_extractor/fip/extractor.py

```
@@ -0,0 +1,322 @@
+
+        return index_data
+
+    def _extract_timing_from_csv(self) -> dict:
+        """
+        Extract session timing from camera metadata CSV files.
+
+        Uses CpuTime column which contains timezone-aware ISO 8601 timestamps,
+        and converts them to the local timezone specified in job_settings.
+
+        Returns
+        -------
+        dict
+            Extracted timing information with 'start_time' and 'end_time' keys
+        """
+        timing_data = {}
+        local_tz = ZoneInfo(self.job_settings.local_timezone)
+
+        # Try to get timing from camera_green_iso_metadata stream
+        metadata_stream = self._get_data_stream("camera_green_iso_metadata")
+        if metadata_stream:
+            metadata = metadata_stream.read()
+            if "CpuTime" in metadata.columns and not metadata.empty:
+                # CpuTime is in timezone-aware ISO 8601 format (UTC)
+                # Convert to local timezone
+                start_utc = datetime.fromisoformat(metadata["CpuTime"].iloc[0])
+                end_utc = datetime.fromisoformat(metadata["CpuTime"].iloc[-1])
+                timing_data["start_time"] = start_utc.astimezone(local_tz)
+                timing_data["end_time"] = end_utc.astimezone(local_tz)
+                return timing_data
+
+        # Fall back to camera_red_metadata
+        if not timing_data:
+            metadata_stream = self._get_data_stream("camera_red_metadata")
+            if metadata_stream:
+                metadata = metadata_stream.read()
+                if "CpuTime" in metadata.columns and not metadata.empty:
+                    start_utc = datetime.fromisoformat(metadata["CpuTime"].iloc[0])
+                    end_utc = datetime.fromisoformat(metadata["CpuTime"].iloc[-1])
+                    timing_data["start_time"] = start_utc.astimezone(local_tz)
+                    timing_data["end_time"] = end_utc.astimezone(local_tz)
+                    return timing_data
```

This is way too much validation. If the contract is not working, then something is wrong with the data and you should let it crash when mapping.

---

No longer applicable as of [21320f0](https://github.com/AllenNeuralDynamics/aind-metadata-extractor/pull/15/commits/21320f0d44540aa5e1804be07f0c910fd168cf01)